### PR TITLE
ENH: Uber-refactor of code style, method names, etc.

### PR DESCRIPTION
**Pull request by oesteban**
Includes a draft implementation of #22.

### CODE REVIEW

File: nitransforms/base.py [hunk 1]
```diff
@@ -150,18 +160,25 @@ def resample(self, moving, order=3, mode='constant', cval=0.0, prefilter=True,
...
+
         """
         moving_data = np.asanyarray(moving.dataobj)
         if output_dtype is None:
             output_dtype = moving_data.dtype
 
+        moving_grid = ImageGrid(moving)
+
+        def _map_indexes(ijk):
+            return moving_grid.inverse.dot(self.map(self.reference.affine.dot(ijk)))
```
Comment (oesteban) [inline nitransforms/base.py]: I think we can reduce this down to defining this concatenation, where the affines of the moving and reference grids are applied on input indexes.

File: nitransforms/nonlinear.py [hunk 2]
```diff
@@ -51,111 +51,46 @@ def __init__(self, field, reference=None):
...
-                             0, -1)
-
-        return coordinates + deltas
+        """
+        if inverse is True:
+            raise NotImplementedError
+        ijk = self.reference.index(x)
+        indexes = np.round(ijk).astype('int')
+        if np.any(np.abs(ijk - indexes) > 0.05):
+            print('Some coordinates are off-grid of the displacements field.',
```
Comment (mgxd) [inline nitransforms/nonlinear.py]: why not use `warnings` here?

File: nitransforms/nonlinear.py [hunk 3]
```diff
@@ -51,111 +51,46 @@ def __init__(self, field, reference=None):
...
+                  file=sys.stderr)
+        indexes = tuple([tuple(i) for i in indexes.T])
```
Comment (mgxd) [inline nitransforms/nonlinear.py]:
```suggestion
        indexes = tuple(tuple(i) for i in indexes.T)
```

File: nitransforms/nonlinear.py (hunk 2 above)
Comment (oesteban) [reply nitransforms/nonlinear.py]: Will do. Thanks!

**Pull request merged**